Exercise_7/Adapters/CountryAdapter.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Android.Content;
+using Android.Support.V7.Widget;
+using Android.Views;
+using Android.Widget;
+using Exercise_7.Extensions;
+using Exercise_7.Interfaces;
+using Exercise_7.Models;
+using Java.Lang;
+
+namespace Exercise_7.Adapters
+{
+    public class CountryAdapter : RecyclerView.Adapter, IItemClickListener, IFilterable
+    {
+        private const int TypeName = 1;
+
+        private const int TypeLetter = 2;
+
+        private List<object> sectionLetterCountries;
+
+        private readonly Context context;
+
+        private readonly List<Country> countriesList;
+
+        private List<Country> originalDatas;
+
+        private static List<object> GetSectionLetterCountriesList(IEnumerable<Country> countries)
+        {
+            var headerLetterCountries = new List<object>();
+            headerLetterCountries.AddRange(countries.OrderBy(x => x.CountryName));
+            var sectionLetter = "";
+            for (var iCount = 0; iCount < headerLetterCountries.Count; iCount++)
+            {
+                var firstLetter = ((Country)headerLetterCountries[iCount]).CountryName[0].ToString();
+                if (sectionLetter.Equals(firstLetter)) continue;
+                sectionLetter = firstLetter;
+                headerLetterCountries.Insert(iCount, firstLetter);
+                iCount++;
+            }
+            return headerLetterCountries;
+        }
+
+        public CountryAdapter(IEnumerable<Country> countries, Context context)
```

Comment:
Let it as `List<Country>` instead of `IEnumerable<Country> `

Reply:
Done